### PR TITLE
feat: create database with options

### LINKED ISSUE #3710: Support creating databases with options
### What problem does the new feature solve?

This feature request is the preparatory work of #3708 

Currently GreptimeDB does not support [creating databases](https://docs.greptime.com/reference/sql/create#create-database) with options. We need to support options in `CREATE DATABASE` statement. 

### What does the feature do?

This feature involves 3 parts:
- Parsing `CREATE DATABASE` statement with options. The related code is:
https://github.com/GreptimeTeam/greptimedb/blob/a52aedec5bc050957dee792bb17a4b045cdbfc4d/src/sql/src/parsers/create_parser.rs#L107-L127
- Storing database options in metasrv. GreptimeDB stores database information in a key-value manner.
https://github.com/GreptimeTeam/greptimedb/blob/c4798d191348d51e13c90fa5c2e3cedf72c2d42a/src/common/meta/src/ddl/create_database.rs#L87-L106
- (Optional) Display database options in `SHOW DATABASES` statement

### Implementation challenges

_No response_

### PR DESCRIPTION
**Pull request by tizee**
I hereby agree to the terms of the [GreptimeDB CLA](https://github.com/GreptimeTeam/.github/blob/main/CLA.md).

## Refer to a related PR or issue link (optional)

- This closes #3710 

## What's changed and what's your intention?

- extends `create database` with options support

```
CREATE DATABASE demo with(ttl='2days')
```

__!!! DO NOT LEAVE THIS BLOCK EMPTY !!!__

Please explain IN DETAIL what the changes are in this PR and why they are needed:

- Summarize your change (**mandatory**)
- How does this PR work? Need a brief introduction for the changed logic (optional)
- Describe clearly one logical change and avoid lazy messages (optional)
- Describe any limitations of the current code (optional)

## Checklist

- [ ]  I have written the necessary rustdoc comments.
- [ ]  I have added the necessary unit tests and integration tests.
- [ ]  This PR does not require documentation updates.

### CODE REVIEW

**Comment by WenyXu** [comment]
please run `make clippy`

**Review by tisonkun — COMMENTED** [review]
Now we implement `Display` for all the statements and will use it on logging statement. Thus I suppose you merge/rebase the main branch and keep the new added options displayed also.

**Comment by tisonkun** [comment]
@tizee I'm glad to review this patch. Would you have time to rebase and add some sqlness tests?

**Comment by tizee** [comment]
> e time to rebase and add some sqlness tests?

Sorry for not replying in time.  @tisonkun Would you please review this PR?

File: tests/cases/standalone/common/create/create_database.sql [hunk 1]
```diff
@@ -1,6 +1,6 @@
 create database illegal-database;
 
-create database 'illegal-database';
+create database 'illegal-database' with (ttl = '1h');
```
Comment (tisonkun) [inline tests/cases/standalone/common/create/create_database.sql:3]: I'd prefer add a new line instead of modifying inline.

**Review by tisonkun — COMMENTED** [review]
Rest LGTM for the parsing part.

But I don't find code related to the retrieving and applying ttl part. Will we implement it later? cc @tizee @v0y4g3r

**Review by evenyag — COMMENTED** [review]
Do we have a way to pass database options to tables in the database?

File: src/operator/src/statement/ddl.rs [hunk 2]
```diff
@@ -702,6 +702,7 @@ impl StatementExecutor {
         catalog: &str,
         database: &str,
         create_if_not_exists: bool,
+        options: Option<HashMap<String, String>>,
```
Comment (v0y4g3r) [inline src/operator/src/statement/ddl.rs:705]: `HashMap` would be enough:

```suggestion
        options: HashMap<String, String>,
```

**Comment by tisonkun** [comment]
I can see that the passing options part is tracked at https://github.com/GreptimeTeam/greptimedb/issues/3708.

Then I think it's OK to only handle the parsing and passing part in this patch, as long as @v0y4g3r track the overall progress.

I'm rebasing this PR and resolve comments.

**Review by tisonkun — APPROVED** [review]

**Comment by killme2008** [comment]
@MichaelScofield @evenyag please take a look.

File: src/common/meta/src/rpc/ddl.rs [hunk 3]
```diff
@@ -645,7 +645,7 @@ pub struct CreateDatabaseTask {
     pub catalog: String,
     pub schema: String,
     pub create_if_not_exists: bool,
-    pub options: Option<HashMap<String, String>>,
+    pub options: HashMap<String, String>,
```
Comment (evenyag) [inline src/common/meta/src/rpc/ddl.rs:648]: @WenyXu Does it have a compatibility issue as structs like `CreateDatabaseTask` derive `Serialize` and `Deserialize`. We may encounter an error if we deserialize the map from `null`.

Comment (WenyXu) [reply src/common/meta/src/rpc/ddl.rs:648]: > @WenyXu Does it have a compatibility issue as structs like `CreateDatabaseTask` derive `Serialize` and `Deserialize`. We may encounter an error if we deserialize the map from `null`.

Let's add the `#[serde_as(deserialize_as = “DefaultOnNull”)]`

Comment (tisonkun) [reply src/common/meta/src/rpc/ddl.rs:648]: Make sense. Confirm that `null` can be a compatibility issue.

Comment (tisonkun) [reply src/common/meta/src/rpc/ddl.rs:651]: Updated.

**Review by evenyag — APPROVED** [review]
LGTM

**Pull request merged**